tests/unit/test_x86.c

```diff
@@ -981,6 +981,110 @@ static void test_x86_nested_uc_emu_start_exits(void)
     OK(uc_close(uc));
 }
 
+static bool test_x86_correct_address_in_small_jump_hook_callback(uc_engine *uc, int type, uint64_t address, int size, int64_t value, void *user_data)
+{
+  // Check registers
+  uint64_t r_rax = 0x0;
+  uint64_t r_rip = 0x0;
+  OK(uc_reg_read(uc, UC_X86_REG_RAX, &r_rax));
+  OK(uc_reg_read(uc, UC_X86_REG_RIP, &r_rip));
+  TEST_CHECK(r_rax == 0x7F00);
+  TEST_CHECK(r_rip == 0x7F00);
+
+  // Check address
+  TEST_CHECK(address == 0x7F00);
+
+  return false;
+}
+
+static void test_x86_correct_address_in_small_jump_hook(void)
+{
+    uc_engine *uc;
+    // movabs $0x7F00, %rax
+    // jmp  *%rax
+    char code[] = "\x48\xb8\x00\x7F\x00\x00\x00\x00\x00\x00\xff\xe0";
+
+    uint64_t r_rax = 0x0;
+    uint64_t r_rip = 0x0;
+    uc_hook hook;
+
+    uc_common_setup(&uc, UC_ARCH_X86, UC_MODE_64, code, sizeof(code) - 1);
+    OK(uc_hook_add(uc, &hook, UC_HOOK_MEM_UNMAPPED, test_x86_correct_address_in_small_jump_hook_callback, NULL, 1, 0));
+
+
+    uc_assert_err(
+        UC_ERR_FETCH_UNMAPPED,
+        uc_emu_start(uc, code_start, code_start + sizeof(code) - 1, 0, 0));
+
+    OK(uc_reg_read(uc, UC_X86_REG_RAX, &r_rax));
+    OK(uc_reg_read(uc, UC_X86_REG_RIP, &r_rip));
+    TEST_CHECK(r_rax == 0x7F00);
+    TEST_CHECK(r_rip == 0x7F00);
```

Comment:
I can't understand why `r_ip==0x7F00`. This jump won't be taken due to the error and the rip should be left unchanged.

Reply:
I think it's ok to do this assert (maybe could be in other test, but is good to be tested).

The point is: what's the expected behaviour? I understood that Unicorn worked as this test describe (IP = Jump's target address).